.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,27 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Pre-Commit
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+
+    # Runs a single command using the runners shell
+    - name: Run a one-line script
+      run: ./pre-commit.sh
```

Comment:
Pre-commit mutates the code, so need to check that it didn't change.  This also seems redundant with some of the checks already there...should we organize a bit better (cc @epicfaace)?